js/ui/components/clubhouse.js

```diff
@@ -703,7 +703,13 @@ class ClubhouseItemBanner extends ClubhouseNotificationBanner {
         super(notification);
         this.actor.add_style_class_name('clubhouse-item-notification');
         this._topBanner = null;
-        callback();
+
+        // We can't call the callback directly because it depends on the object
+        // constructed here
+        GLib.idle_add(GLib.PRIORITY_DEFAULT, () => {
+            callback();
```

Comment:
Why do we need to pass to it a callback? I think we can remove that callback argument and move its body here. If I am correct the only place where we are using tha callback is [here](https://github.com/endlessm/gnome-shell/blob/master/js/ui/components/clubhouse.js#L1011-L1013). Or if you want to keep the callback, we can even pass `this` as the first argument of the callback.

Reply:
Yes, we can remove the callback in this case and just call the code just below, but I think that this callback is there to have the same interface than the quest banner. Maybe it's better to add this as parameter, as you said, so we don't need the idle_add and will avoid future problems.

Reply:
I've changed this to pass `this` as a param to the callback.